src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,125 @@
 package core.basesyntax;
 
-/**
+import java.util.NoSuchElementException;
+
+/*
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
+
 public class ArrayList<T> implements List<T> {
+    private static final int CAPASITY = 10;
+    private T[] arrayGeneric;
+    private int size;
+
+    public ArrayList() {
+        arrayGeneric = (T[]) new Object[CAPASITY];
+    }
 
     @Override
     public void add(T value) {
-
+        if (size == arrayGeneric.length) {
+            arrayGeneric = resize();
+        }
+        arrayGeneric[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        }
+        if (index > size || index < 0) {
+            throw new ArrayIndexOutOfBoundsException(
+                    "This index is not suitable for the given array");
+        }
```

Comment:
this check is all right
but on line 29 you already cover the case if `index == size`
so, here it is possible to just call `checkIndex()` instead of lines 33-36

is it clear?

Reply:
I'm not sure, because Dmytro yesterday on QA  told me didn't touch this "method public void add(T value, int index)"  like here we have other logic that why I didn't touch this method with the same exception?!